config/routes.rb

```diff
@@ -21,6 +21,18 @@
         post "/releases/:version/rollback", to: "releases#rollback"
       end
 
+      resources :heritages, except: [:new, :edit] do
+        post   :env_vars, on: :member, to: "heritages#set_env_vars"
+        delete :env_vars, on: :member, to: "heritages#delete_env_vars"
+
+        post "/trigger/:token", to: "heritages#trigger"
+        resources :oneoffs, only: [:show, :create]
+
+        get "/releases", to: "releases#index"
+        get "/releases/:version", to: "releases#show"
+        post "/releases/:version/rollback", to: "releases#rollback"
+      end
```

Comment:
If you expand the top you'll find nearly exactly the same code. Is this a duplicate?

Reply:
you mean I can do like below?

```ruby
[true, false].each do |shallow|
  resources :heritages, shallow: shallow, except: [:new, :edit] do
    # ...
  end
end